src/app/beneficiaries/create.jsx

```diff
@@ -1,192 +1,441 @@
 'use client'
-import Phone from '../components/icons/phone'
-import Location2 from '../components/icons/location-2'
-import User from '../components/icons/user'
-import UserLaptop from '../components/icons/user-laptop'
-import Clipboard from '../components/icons/clipboard'
-import Identification from '../components/icons/id'
-import Gender from '../components/icons/gender'
 /* eslint-disable no-unused-vars */
 import React from 'react'
 /* eslint-enable no-unused-vars */
+import { useRouter } from 'next/navigation'
+import axios from 'axios'
 
-// const axios = require('axios').default
+export function getCurrentDate() {
+	const currentDate = new Date()
+	const year = currentDate.getFullYear()
+	const month = String(currentDate.getMonth() + 1).padStart(2, '0') // Months are zero-based
+	const day = String(currentDate.getDate()).padStart(2, '0')
 
-export default function CreateModal({ isVisible, onClose }) {
-	if (!isVisible) return null
+	return `${year}-${month}-${day}`
+}
 
-	const toClose = x => {
-		if (x.target.id === 'close') {
-			onClose()
-		}
-	}
+function CreateModal({ closeModal }) {
+	const router = useRouter()
+
+	const formattedDate = getCurrentDate()
 
-	// const router = useRouter()
 	async function onSubmit(event) {
 		event.preventDefault()
-		// TODO: waiting for creation API implementation
-		// const formData = new FormData(event.target)
-		// axios
-		// 	.post(
-		// 		'https://65dc59f1e7edadead7ebb34d.mockapi.io/api/v1/beneficiary',
-		// 		formData
-		// 	)
-		// 	.then(function (response) {
-		// 		// Navigate to the newly created beneficiary
-		// 		router.push('/beneficiaries/' + response.data.id.toString())
-		// 	})
-		// 	.catch(function (error) {
-		// 		// TODO: handle error
-		// 		console.log(error)
-		// 	})
+		const formData = new FormData(event.target)
+		// WARNING this might be deleted in future development
+		formData.append('registration_date', formattedDate)
+		formData.append('age', '0')
+		// up until here
+		const object = {}
+		formData.forEach((value, key) => (object[key] = value))
+		const json = JSON.stringify(object)
+		axios
+			.post(process.env.NEXT_PUBLIC_BASE_URL + '/acat/patient', json, {
+				headers: {
+					'Content-Type': 'application/json'
+				}
+			})
+			.then(function (response) {
+				// Navigate to the newly created beneficiary
+				closeModal()
+				router.push('/beneficiaries/' + response.data.id.toString())
+			})
+			.catch(function (error) {
+				alert(
+					`Ha habido un error al crear al nuevo beneficiario: ${error.response.data.detail}`
```

Comment:
Creating a beneficiary with invalid data will result in this error. It's not very descriptive:
![image](https://github.com/ISPP-07/frontend-acat/assets/72880234/39a0ef4c-36d4-47fe-a2d5-1d2bf3d697ad)